### PR TITLE
Personal user space and CRKBD R2G keymap

### PR DESCRIPTION
**Pull request by RMeli**
Personal user space and keymap for CRKBD R2G.

## Description

Added Unicode keys and useful tap dances (parenthesis, shift, caps lock).

Added personal keymap for the CRKBD (R2G version).

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/crkbd/keymaps/rmeli/keymap.c [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#include QMK_KEYBOARD_H
+#include <stdio.h>
```
Comment (drashna) [inline keyboards/crkbd/keymaps/rmeli/keymap.c:21]:
```suggestion
```

File: keyboards/crkbd/keymaps/rmeli/keymap.c [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+// + ------- +
+// + KEY MAP |
+// + ------- +
+
+// Layer names
+#define _QWERTY 0
+#define _COLEMAK_DH 1
+#define _DWN 2
+#define _UP 3
+#define _CONFIG 4
```
Comment (drashna) [inline keyboards/crkbd/keymaps/rmeli/keymap.c:63]: We do recommend enums for this, as it's easier to maintain,

File: users/rmeli/keyrecords/tap_dances.c [hunk 3]
```diff
@@ -0,0 +1,122 @@
+#include "keyrecords/tap_dances.h"
```
Comment (drashna) [inline users/rmeli/keyrecords/tap_dances.c:1]: Could you add a GPL2+ compatible license header here? 

For instance:
```c
// Copyright 2021 Your Name (@yourgithub)
// SPDX-License-Identifier: GPL-2.0-or-later
```

Or 

https://github.com/qmk/qmk_firmware/blob/6b299fab6e6a3143c061b9ea005253361fcd524b/data/templates/base/keyboard.c#L1-L15

File: users/rmeli/keyrecords/tap_dances.h [hunk 4]
```diff
@@ -0,0 +1,53 @@
+#pragma once
```
Comment (drashna) [inline users/rmeli/keyrecords/tap_dances.h:1]: Could you add a GPL2+ compatible license header here? 

For instance:
```c
// Copyright 2021 Your Name (@yourgithub)
// SPDX-License-Identifier: GPL-2.0-or-later
```

Or 

https://github.com/qmk/qmk_firmware/blob/6b299fab6e6a3143c061b9ea005253361fcd524b/data/templates/base/keyboard.c#L1-L15

File: users/rmeli/keyrecords/unicode.h [hunk 5]
```diff
@@ -0,0 +1,68 @@
+#pragma once
```
Comment (drashna) [inline users/rmeli/keyrecords/unicode.h:1]: Could you add a GPL2+ compatible license header here? 

For instance:
```c
// Copyright 2021 Your Name (@yourgithub)
// SPDX-License-Identifier: GPL-2.0-or-later
```

Or 

https://github.com/qmk/qmk_firmware/blob/6b299fab6e6a3143c061b9ea005253361fcd524b/data/templates/base/keyboard.c#L1-L15

File: users/rmeli/oled/oled.c [hunk 6]
```diff
@@ -0,0 +1,79 @@
+#include "oled/oled.h"
```
Comment (drashna) [inline users/rmeli/oled/oled.c:1]: Could you add a GPL2+ compatible license header here? 

For instance:
```c
// Copyright 2021 Your Name (@yourgithub)
// SPDX-License-Identifier: GPL-2.0-or-later
```

Or 

https://github.com/qmk/qmk_firmware/blob/6b299fab6e6a3143c061b9ea005253361fcd524b/data/templates/base/keyboard.c#L1-L15

File: users/rmeli/oled/oled.h [hunk 7]
```diff
@@ -0,0 +1,9 @@
+#pragma once
```
Comment (drashna) [inline users/rmeli/oled/oled.h:1]: Could you add a GPL2+ compatible license header here? 

For instance:
```c
// Copyright 2021 Your Name (@yourgithub)
// SPDX-License-Identifier: GPL-2.0-or-later
```

Or 

https://github.com/qmk/qmk_firmware/blob/6b299fab6e6a3143c061b9ea005253361fcd524b/data/templates/base/keyboard.c#L1-L15

File: users/rmeli/rmeli.h [hunk 8]
```diff
@@ -0,0 +1,11 @@
+#ifdef UNICODEMAP_ENABLE
```
Comment (drashna) [inline users/rmeli/rmeli.h:1]: Could you add a GPL2+ compatible license header here? 

For instance:
```c
// Copyright 2021 Your Name (@yourgithub)
// SPDX-License-Identifier: GPL-2.0-or-later
```

Or 

https://github.com/qmk/qmk_firmware/blob/6b299fab6e6a3143c061b9ea005253361fcd524b/data/templates/base/keyboard.c#L1-L15

**Review by drashna — CHANGES_REQUESTED** [review]

**Comment by RMeli** [comment]
Thank you @drashna for having a look at this! I think I addressed all the issues.

**Review by drashna — APPROVED** [review]

**Review by noroadsleft — APPROVED** [review]

**Comment by noroadsleft** [comment]
Thanks!

For future reference, we recommend against committing to your `master` branch as you've done here, because pull requests from modified `master` branches can make it more difficult to keep your QMK fork updated. It is highly recommended for QMK development – regardless of what is being done or where – to keep your master updated, but **NEVER** commit to it. Instead, do all your changes in a branch (branches are basically free in Git) and issue PRs from your branches when you're developing.

There are instructions on how to keep your fork updated here:

[**Best Practices: Your Fork's Master: Update Often, Commit Never**](https://docs.qmk.fm/#/newbs_git_using_your_master_branch)

[Fixing Your Branch](https://docs.qmk.fm/#/newbs_git_resynchronize_a_branch) will walk you through fixing up your `master` branch moving forward. If you need any help with this just ask.

Thanks for contributing!

**Comment by RMeli** [comment]
Not sure why I worked directly in `master` here, I switched to branches for other keyboards.Many thanks @noroadsleft for the review and merging.

**Pull request merged**